pkg/ebpf is not buildable on non-glibc (e.g. musl) environments

**Describe what happened:**

Building `system-probe` on Alpine Linux (musl-libc based) fails by the following errors.
```
# github.com/DataDog/datadog-agent/pkg/ebpf/compiler
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: $WORK/b758/_x002.o: in function `__wrap_exp':
compiler.cgo2.c:(.text+0x1): undefined reference to `exp@GLIBC_2.2.5'
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: $WORK/b758/_x002.o: in function `__wrap_log':
compiler.cgo2.c:(.text+0x6): undefined reference to `log@GLIBC_2.2.5'
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: $WORK/b758/_x002.o: in function `__wrap_pow':
compiler.cgo2.c:(.text+0xb): undefined reference to `pow@GLIBC_2.2.5'
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: $WORK/b758/_x002.o: in function `__wrap_log2':
compiler.cgo2.c:(.text+0x10): undefined reference to `log2@GLIBC_2.2.5'
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: $WORK/b758/_x002.o: in function `__wrap_log2f':
compiler.cgo2.c:(.text+0x15): undefined reference to `log2f@GLIBC_2.2.5'
collect2: error: ld returned 1 exit status
```

`pkg/ebpf/compiler` seems wrapping some glibc functions (by this LDFLAGS: https://github.com/DataDog/datadog-agent/blob/d1c5de31e1bba72dfac459aed5ff9562c3fdcc20/pkg/ebpf/compiler/compiler.go#L10) even if it is built with non-glibc libc (like musl).

I think https://github.com/DataDog/datadog-agent/blob/master/pkg/ebpf/compiler/wrapper.h should be switched by `#ifdef __GLIBC__ `.

I'm currently working around this error by applying a patch.
https://github.com/seqsense/datadog-agent-alpine/pull/25/files#diff-9f7490cb5f442f98432f2fd3c530e509b2822925a7f121e50e9d3d5cd116375aR65-R94

**Describe what you expected:**

`system-probe` is buildable on Alpine Linux.

**Steps to reproduce the issue:**

Dockerfile to build datadog-agent for Alpine Linux: https://github.com/seqsense/datadog-agent-alpine/blob/e2d66a47a6157689b9e7bcd2ec917e7a467f7f52/Dockerfile
Easy way to build using this version of Dockerfile:
```shell
git clone https://github.com/seqsense/datadog-agent-alpine.git -b renovate/datadog
cd datadog-agent-alpine/
git checkout e2d66a47a6
make ENABLE_PROCESS_AGENT=1 ENABLE_SECURITY_AGENT=1 ENABLE_TRACE_AGENT=1 ENABLE_SYSTEM_PROBE=1
```

Result on CI: https://github.com/seqsense/datadog-agent-alpine/runs/1962043223#step:6:2221
Related: https://github.com/seqsense/datadog-agent-alpine/pull/25

--- a/pkg/ebpf/compiler/wrapper.h
+++ b/pkg/ebpf/compiler/wrapper.h
@@ -13,6 +13,8 @@ $ objdump -p bin/system-probe/system-probe
 $ nm bin/system-probe/system-probe | grep GLIBC_X.XX
 */
 
+#ifdef __GLIBC__
+
 #ifdef __x86_64__
 #define GLIBC_VERS "GLIBC_2.2.5"
 #elif defined(__aarch64__)
@@ -47,6 +49,34 @@ asm(".symver __" #func "_prior_glibc, " #func "@" GLIBC_VERS);  \
 float __wrap_ ## func (float x) {                               \
   return __ ## func ## _prior_glibc(x);                         \
 }
+
+#else
+
+// Use functions directly for non-GLIBC environments.
+
+#define symver_wrap_d1(func)                                    \
+double func(double x);                                          \
+                                                                \
+double __wrap_ ## func (double x) {                             \
+  return func(x);                                               \
+}
+
+#define symver_wrap_d2(func)                                    \
+double func(double x, double y);                                \
+                                                                \
+double __wrap_ ## func (double x, double y) {                   \
+  return func(x, y);                                            \
+}
+
+#define symver_wrap_f1(func)                                    \
+float func(float x);                                            \
+                                                                \
+float __wrap_ ## func (float x) {                               \
+  return func(x);                                               \
+}
+
+#endif
+
 
 symver_wrap_d1(exp)
 symver_wrap_d1(log)

--- a/releasenotes/notes/make-ebpf-compiler-buildable-on-non-glibc-env-4ebe7f98ea1ace2c.yaml
+++ b/releasenotes/notes/make-ebpf-compiler-buildable-on-non-glibc-env-4ebe7f98ea1ace2c.yaml
@@ -1,0 +1,11 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Make ebpf compiler buildable on non-GLIBC environment.

